tests/Http/Role/AddConstraintRequestHandlerTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Http\Role;
+
+use Broadway\CommandHandling\Testing\TraceableCommandBus;
+use CultuurNet\UDB3\Http\ApiProblem\ApiProblem;
+use CultuurNet\UDB3\Http\ApiProblem\AssertApiProblemTrait;
+use CultuurNet\UDB3\Http\ApiProblem\SchemaError;
+use CultuurNet\UDB3\Http\Request\Psr7RequestBuilder;
+use CultuurNet\UDB3\Http\Response\AssertJsonResponseTrait;
+use CultuurNet\UDB3\Model\ValueObject\Identity\UUID;
+use CultuurNet\UDB3\Role\Commands\AddConstraint;
+use CultuurNet\UDB3\Role\ValueObjects\Query;
+use Fig\Http\Message\StatusCodeInterface;
+use PHPUnit\Framework\TestCase;
+use Slim\Psr7\Response;
+
+class AddConstraintRequestHandlerTest extends TestCase
+{
+    use AssertApiProblemTrait;
+    use AssertJsonResponseTrait;
+
+    private AddConstraintRequestHandler $handler;
+
+    private TraceableCommandBus $commandBus;
+
+    protected function setUp()
+    {
+        $this->commandBus = new TraceableCommandBus();
+        $this->handler = new AddConstraintRequestHandler($this->commandBus);
+    }
+
+    /**
+     * @test
+     */
+    public function it_throws_when_query_is_not_given(): void
+    {
+        $roleId = '1ed1588b-a771-44ce-bac0-8f19f09a7d0f';
+        $request = (new Psr7RequestBuilder())
+            ->withRouteParameter('id', $roleId)
+            ->withJsonBodyFromArray([])
+            ->build('POST');
+
+        $this->commandBus->record();
```

Comment:
The record is not needed here.

Reply:
I'm testing if the command bus didn't receive commands, so it should record or it would always return empty, even if it dispatched commands. I've moved the record step to the setup function, as I think it shouldn't be part of an individual test case but just a given.